src/Compiler/TypedTree/tainted.fs

```diff
@@ -101,11 +101,12 @@ type internal Tainted<'T> (context: TaintedContext, value: 'T) =
             | :? TypeProviderError -> reraise()
             | :? AggregateException as ae ->
                     let errNum,_ = FSComp.SR.etProviderError("", "")
-                    let messages = [for e in ae.InnerExceptions -> e.Message]
+                    let messages = [for e in ae.InnerExceptions -> if isNull e.InnerException then e.Message else (e.Message + ": " + e.GetBaseException().Message)]
```

Comment:
Apologies for the confusion, I think I was wrong about this line before, sorry about that. This won't behave as you (and I) have come to expect it.

Here, we cannot rely on the virtual method `GetBaseException()` for two reasons: (1) it will returns the _inner most exeption_ by definition, which can lead to this returning multiple times the same exception. And (2) it may be overridden.

Other than the single-exception case below, we can just simply use what was already there:
```suggestion
                    let messages = [for e in ae.InnerExceptions -> e.Message]
```

That is, I think the new code is actually leading to hiding errors by repeating the same (deepest) exception message. See line 42, which takes the sequence of errors into account already, so we don't have to check the inner exception field here:

```f#
    override _.Message = 
        match errors with
        | [text] -> text
        | inner -> 
            // imitates old-fashioned behavior with merged text
            // usually should not fall into this case (only if someone takes Message directly instead of using Iter)
            inner            
            |> String.concat Environment.NewLine
```

Reply:
Are you saying we should do something like "if (inner is null) or (message = inner_message) then just message else (message and inner_message)"?

Reply:
No, this one just gets all inner messages through that property, and if that happens, existing code already concatenates messages properly. Nothing we need to do here. 

The other line you fixed, is about a single error, and the inner exception was ignored. You can treat that like this one (a list), or like the suggested code. 

In hindsight, maybe both should just use the list argument. I can make a PR against your branch to show what I mean. Sorry if I'm being unclear.

Reply:
I don't care the ex.InnerException: The information needed for the user is in ex.InnerException.InnerException.InnerException.InnerException.Message

Reply:
You are constantly talking about exception list, but I care the nested stack, not the list.
Have a look at this:

```fsharp
open System
let myEx = AggregateException(Exception("new", Exception("new", ArgumentException("hi"))));;
myEx.InnerExceptions.Count
// val it: int = 1
// The "hi" is hiding. You can't see it from looping the list.

myEx.GetBaseException().Message
// val it: string = "new"
```

Reply:
really, what is bad about `string ex`?

> val it: string =
  "System.AggregateException: One or more errors occurred. (new)
 ---> System.Exception: new
 ---> System.Exception: new
 ---> System.ArgumentException: hi
   --- End of inner exception stack trace ---
   --- End of inner exception stack trace ---
   --- End of inner exception stack trace ---"

Reply:
I'm trying to fix the real user issue here, which is that the current meaningful message (most inner ex.Message) is hidden. 
I get it that AggregateException may in theory have a task that contains multiple exceptions (e.g. you can start multiple tasks parallel, then Task.WaitAll and there might be many), and some may override whatever method in theory, and in theory there might be an exception in the world that is not covered yet. However in a context of F#, task { ... }, and failwith, this is not typical. I would really like to fix the real user issue. Which is accepting this PR and then go to the better direction step by step: if there are other cases not captured, cover them by a next PR.

Reply:
I agree. But I think you are missing the feature that's already there: it shows all exceptions when there's an AggregateException. Your change hides these, so it's a step back. 

The inverse is true when it's not an AggregateException. In that case, it shows the top and the bottom of the stack, which is an improvement. 

I'm not trying to block you. I'm trying to show you what the impact is of your change. I doubt your intent was to show less, unless I've misunderstood your original issue. 

(and I'm fully aware of PRs never making it, some of my own ended up there)

Reply:
It would be true if it would be `ae.InnerException` but my code says `e.InnerException` so it's going loop all AggregateException's exceptions and getting their inner-exceptions.